cli/crates/server/src/event.rs

```diff
@@ -1,23 +1,26 @@
-use std::path::PathBuf;
+use std::path::{Path, PathBuf};
 use tokio::sync::broadcast::Receiver;
 
 /// server lifecycle related events
 #[derive(Clone, Debug, PartialEq, Eq)]
 pub enum Event {
-    /// emitted when a schema change is detected
+    /// emitted when a schema or UDF change is detected
     /// and the server should be reloaded
     Reload(PathBuf),
     /// emitted when the bridge is ready to receive requests
     BridgeReady,
     /// emitted when the proxy server has a startup error
     ProxyError,
+    /// emitted when an SDL schema produced from the TS config has been written. Contains the file
+    /// path where it was written.
+    NewSdlFromTsConfig(Box<Path>),
```

Comment:
Why `Box<Path>` and not `PathBuf`? Just curious.

Reply:
Like `Box<str>` vs `String`:

```
[src/main.rs:3] std::mem::size_of::<Box<Path>>() = 16
[src/main.rs:4] std::mem::size_of::<PathBuf>() = 24
```

`Box<Path>` doesn't have the `capacity` field. It's more efficient if you're not planning to mutate it.

Reply:
That is the best reason for me. Nice.